File: ch14/hello_single.c
#include<stdio.h>
#include<unistd.h>

#define NUM 5

void print_msg(char *);

int main(int argc, char const *argv[])
{
  print_msg("hello ");
  print_msg("world\n");
  return 0;
}

void print_msg(char *m){
  for (size_t i = 0; i < NUM; i++)
  {
    printf("%s",m);
    fflush(stdout);
    sleep(1);
  }
  
}

File: ch14/ch14_src/src/bin/hello_single.rs
use std::{io::{stdout, Write}, thread::sleep, time::Duration};

const NUM:u128 = 5;

fn main(){
  print_msg("hello ");
  print_msg("world\n");
}

fn print_msg(s:&str){
  for _ in 0..NUM {
      print!("{}",s);
      let _ = stdout().flush();
      sleep(Duration::from_secs(1));
  }
}